app/services/chat_channels/create_with_users.rb

```diff
@@ -0,0 +1,43 @@
+module ChatChannels
+  class CreateWithUsers
+    def initialize(users: [], channel_type: "direct", contrived_name: "New Channel", membership_role: "member")
+      @users = users
+      @channel_type = channel_type
+      @contrived_name = contrived_name
+      @membership_role = membership_role
+    end
+
+    attr_reader :users, :channel_type, :membership_role
+    attr_accessor :contrived_name
+
+    def self.call(*args)
+      new(*args).call
+    end
+
+    def call
+      raise "Invalid direct channel" if invalid_direct_channel?(users, channel_type)
+
+      usernames = users.map(&:username).sort
+      slug = if channel_type == "direct"
+               usernames.join("/")
+             else
+               "#{@contrived_name.to_s.parameterize}-#{rand(100_000).to_s(26)}"
+             end
+
+      contrived_name = channel_type == "direct" ? "Direct chat between " + usernames.join(" and ") : @contrived_name
```

Comment:
We have a problem here with variable `contrived_name` without @ this is `null`. The most thing strange is in the class `initialize` the value is normal, but after that the value is null

Reply:
I think the problem here is that the class defines an `attr_accessor` and then here there's a local variable with the same name of the accessor. You need the following:

- `attr_reader :contrived_name`
- this local variable to have a different name